### PR TITLE
Obs client

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
Script to fetch controller key and state variables
Creates var file for ansible in playbooks/vars folder

### CODE REVIEW

**Review by lego963 — DISMISSED** [review]

File: playbooks/files/obs_cli.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+def generate_vars_file(state, key_path):
+    inv_output = {
+        'controller_state': {}
+    }
+    variables = inv_output['controller_state']
+    variables.update({'controller_key': key_path})
+    for outputs in get_instances_info(state):
+        variables.update(outputs)
+    if variables:
+        root_path = os.path.abspath(f'{os.path.dirname(__file__)}/../..')
```
Comment (outcatcher) [inline playbooks/files/obs_cli.py]: Can be made a constant

File: playbooks/files/obs_cli.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+
+def generate_vars_file(state, key_path):
+    inv_output = {
+        'controller_state': {}
+    }
+    variables = inv_output['controller_state']
+    variables.update({'controller_key': key_path})
+    for outputs in get_instances_info(state):
+        variables.update(outputs)
```
Comment (outcatcher) [inline playbooks/files/obs_cli.py]: Variables will be overwritten if there are overlapping keys

File: playbooks/files/obs_cli.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    with open(state_file) as s_file:
+        return json.load(s_file)
+
+
+def generate_vars_file(state, key_path):
+    inv_output = {
+        'controller_state': {}
+    }
+    variables = inv_output['controller_state']
+    variables.update({'controller_key': key_path})
```
Comment (outcatcher) [inline playbooks/files/obs_cli.py]: `{'controller_key': key_path}` can be moved in dictionary declaration

File: playbooks/files/obs_cli.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    """Check if local file is not up to date with remote"""
+    if not os.path.isfile(file_name):
+        return True
+    with open(file_name, 'rb') as trg_file:
+        md5 = hashlib.md5(trg_file.read()).hexdigest()
+    return remote_md5 != md5
+
+
+def get_item_from_s3(file, item_name, credential: Credential) -> str:
+    """Download existing key from s3 or create a new one and upload"""
```
Comment (outcatcher) [inline playbooks/files/obs_cli.py]: Please change docstring, the method is used not only for the key now

File: playbooks/files/obs_cli.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                      aws_secret_access_key=credential.secret,
+                      aws_session_token=credential.security_token)
+
+    obs = session.resource('s3', endpoint_url=S3_ENDPOINT)
+    bucket = obs.Bucket(BUCKET)
+    try:
+        file_md5 = bucket.Object(item_name).e_tag[1:-1]
+    except ClientError as cl_e:
+        if cl_e.response['Error']['Code'] == '404':
+            print('The object does not exist in s3. Generating new one...')
```
Comment (outcatcher) [inline playbooks/files/obs_cli.py]: Is this print valid?

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by Polina-Gubina — DISMISSED** [review]

**Review by lego963 — DISMISSED** [review]

**Review by Polina-Gubina — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Review by outcatcher — APPROVED** [review]

**Pull request merged**